File: src/main/java/com/dillian/initiateservice/services/GameCreationService.java
package com.dillian.initiateservice.services;

import com.dillian.initiateservice.dtos.*;
import com.dillian.initiateservice.util.BuildingIds;
import com.dillian.initiateservice.util.StartingValues;
import lombok.AllArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;

import java.util.*;

@Service
@AllArgsConstructor
@Slf4j
public class GameCreationService {

    private final DistrictCreationService districtCreationService;

    public InitiateDTO assembleInitiateDTO(SupervisorDTO supervisor) {
        InitiateDTO dto = new InitiateDTO();
        dto.setId(1L);
        dto.setFunds(StartingValues.FUNDS_STARTING_AMOUNT);
        dto.setPopularity(StartingValues.POPULARITY_STARTING_AMOUNT);
        dto.setResearch(StartingValues.RESEARCH_STARTING_AMOUNT);
        dto.setBuildingRequests(districtCreationService.selectStartingBuildings());
        dto.setSupervisor(supervisor);
        List<Tile> tiles = districtCreationService.generateStartingTiles();
        districtCreationService.assignBuildingsToTiles(tiles);
        dto.setTiles(tiles);
        dto.setDistricts(districtCreationService.groupTilesIntoDistricts(tiles));
        return dto;
    }
}

File: src/main/java/com/dillian/initiateservice/services/GameCreationService.kt
package com.dillian.initiateservice.services

import com.dillian.initiateservice.dtos.*
import com.dillian.initiateservice.util.*
import org.springframework.stereotype.Service

@Service
class GameCreationService(
    private val tileCreationService: TileCreationService,
    private val buildingCreationService: BuildingCreationService
) {

    fun buildInitiateDTO(supervisor: SupervisorDTO): GameDTO {
        return GameDTO(
            1L,
            0,
            0,
            0,
            FUNDS_STARTING_AMOUNT,
            0,
            0,
            0,
            POPULARITY_STARTING_AMOUNT,
            0,
            RESEARCH_STARTING_AMOUNT,
            0,
            0,
            0,
            supervisor,
            assignBuildingsToTiles(),
            "",
            ""
        )
    }

    fun assignBuildingsToTiles(): List<Tile> {
        val tiles: List<Tile> = tileCreationService.generateStartingTiles()
        val buildingsWithTiles = buildingCreationService.createBuildingsInGame()
        val buildingsByTileId = buildingsWithTiles.associate { (building, tileId) -> tileId to building }
        tiles.forEach { tile ->
            tile.building = buildingsByTileId[tile.id]
        }
        return tiles
    }
}